### ISSUE TITLE
Setting toggle: improved decision criteria, documentation and UI

### ISSUE THREAD

**Issue by heydouble-a** [Shopify/polaris-react-archive#8328]
## Feature request summary

1. [Improve to the setting toggle component](https://www.figma.com/file/6DZzPcdCuMqTogvzZAveBW/Split-customer-account-settings?node-id=1851%3A20110&t=DHX7fHu1rjE6jh5p-1) to make it a more viable component for designers to use and achieve good UX.
2. Improve Polaris documentation to clarify when- and how-to-use, utilizing decision criteria that makes it easy for designers to decide when to use a setting toggle.

## Rationale

The Polaris decision criteria for when to use a [setting toggle](https://polaris.shopify.com/components/setting-toggle) vs. [checkbox](https://polaris.shopify.com/components/checkbox) is unclear.

The current setting toggle component and guidelines can easily lead to bad UX. 
For example:
- Using one string to indicate the setting name and status is difficult to scan.
- The primary green “Activate” CTA can easily be mistaken for the setting being “on”.
---

## Proposal
[Setting toggle proposal](https://docs.google.com/document/d/1E-71SFvjDPaAtbpMbZyFXgR9Ad_i-s4cvCHVTPy8Izk/edit#) (Google Doc)

---
🌟 Feature requests that are not yet planned will be closed. We then use the issue’s :+1: upvotes to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

**Comment by heydouble-a** [comment]
@oluwatimio Could you run a check to see where the existing setting toggle is used in **admin**? I want to make sure we cover off on all use cases, and then later will use this list to reach out to respective UX teams about the improvements.

**Comment by oluwatimio** [comment]
> @oluwatimio Could you run a check to see where the existing setting toggle is used in **admin**? I want to make sure we cover off on all use cases, and then later will use this list to reach out to respective UX teams about the improvements.

Hey @heydouble-a , for the admin there are two places where I have been able to spot the current `SettingToggle` being used

1. The internal org admin at `https://shopify.plus/internal/settings`

<img width="1679" alt="Screenshot 2023-02-15 at 12 02 48 PM" src="https://user-images.githubusercontent.com/13557893/219101053-7449d9d9-4c50-4214-8dc3-d49b8f7d5556.png">

2. Preferences section / Customer privacy section of online store channel

<img width="956" alt="Screenshot 2023-02-15 at 12 04 30 PM" src="https://user-images.githubusercontent.com/13557893/219101273-4dc9ad23-0fcd-48c6-98f2-ff39b689a9b7.png">

**Comment by oluwatimio** [comment]
Apart from ^ there are other 1P apps that use the Settings Toggle

**Comment by heydouble-a** [comment]
Okay... I think it would be worthwhile to know how our 1P apps are using this. Would you send that as well?

**Comment by oluwatimio** [comment]
> Okay... I think it would be worthwhile to know how our 1P apps are using this. Would you send that as well?

I don't think we necessarily need to check all 1P apps. There are a ton of apps using it as seen here https://grokt.shopify.io/repos?q=%3CSettingToggle and probably 3P apps too. Im not sure Installing all of them and checking how they are using the toggle may be worthwhile 🤔. We can definitely communicate these SettingsToggle changes in a changelog when updating Polaris

**Comment by giliaangell** [comment]
@heydouble-a I'd love to help support this--those proposed designs look familiar! ; ) The activate/deactivate and on/off terminology is another discussion going on across the content design community. @tommcgathshopify and I worked through all the language in our Payments Settings to update it (for the new stack Settings in France and soon, Romania). We're working to remove the non-inclusive language of "enabled/disabled." 

Can I add myself to the assignees?

**Comment by heydouble-a** [comment]
Absolutely. I'd love to have you be a part of making this better, and get it right before shipping. cc @deannatron for content on our side

**Comment by jjgali** [comment]
Thank you for putting this together! I love the proposed guidelines for the setting toggle, and the decision tree you've included is very helpful. 

Something you did here that we've failed to do in the past is distinguish between on/off settings and other binary settings options are are **not** simply on/off. 
<img width="1158" alt="Screenshot 2023-02-17 at 10 13 55 AM" src="https://user-images.githubusercontent.com/77791660/219707471-ccfdd1d7-950b-4ece-99a1-a6ff98a20853.png">

When we were making an effort to find language that could replace `enable`/`disable`, we settled on `activate`/`deactivate` because we felt it could include the wide array of binary settings options we were being presented with. But, by naming the card the setting, employing the status badge, and making the setting toggle _exclusive_ to on/off, you've greatly reduced the cognitive load and complexity. Now we can confidently use `Turn on`/`Turn off` for setting toggles.

**Comment by tommcgathshopify** [comment]
Hey all, I've been working on some logic here to distinguish between when to specifically use `On/Off` and `Activate/Deactivate` since it is not as clear cut as it used to be with `enable/disable.`

Does this model give guidance on the language or is it from the UI side? From what I can see, I haven't been able to find guidance on what to use. For me, simple settings should be `On/Off` as this is analogue to `enable/disable`, and larger stand-alone features can use `activate/deactivate`. 

Would love to see some input on this. There's also a bit of a question around when to use `inactive` vs. `active.`

**Comment by jjgali** [comment]
Hey @tommcgathshopify! I think the changes suggested here would allow you to write logic that is more reminiscent of what you had with `enable/disable`. These proposed changes would mean that the setting toggle would only support binary options that are simply On/Off. Other binary options should seek alternative UI, such as a check box or radio button. You can see that illustrated in the decision tree above.

**Comment by heydouble-a** [comment]
[Polaris documentation PR here](https://github.com/Shopify/polaris/pull/8497).

**Comment by Jen-Bader** [comment]
Hi @heydouble-a et al., 
2 QQs: 
1. Is the intent that this would replace the Visible/Hidden settings used for blog posts and the theme editor? 

I ask b/c the Inbox team took our lead from this standard in reference when giving merchants the option to make the online store chat widget visible to customers, and also for the instant answers portion of the interface. <img width="843" alt="image" src="https://user-images.githubusercontent.com/55419135/221918655-959360c9-b889-48ff-95ca-efe22bf6aa72.png">
<img width="263" alt="image" src="https://user-images.githubusercontent.com/55419135/221925468-6abd254f-52c8-4bbf-86dd-435f0eb5bce6.png">


2. Would this apply to checkbox/notification settings across the admin? e.g., in admin settings, the checkbox is used for "on" status. In this case there are 2 settings, both are binary. Although there is a progressive order here, I don't think of this as  "progressive disclosure" in the proper sense b/c it doesn't involve another screen. 
<img width="955" alt="image" src="https://user-images.githubusercontent.com/55419135/221921079-811b85a7-5090-4d05-8dcf-d287de0a1db6.png">

If the above scenario would switch to on/off, want to show an example where it wouldn't fit, based on my understanding of this update, which is our checkbox setting (b/c it is not a standalone setting).
<img width="768" alt="image" src="https://user-images.githubusercontent.com/55419135/221926584-1a7c6505-d95e-4ee9-8b75-860facd07386.png">


We're making some admin UX improvements, and would like to adopt this in our next release if it's going to be the standard. Please advise. Thanks!

**Comment by heydouble-a** [comment]
Hey @Jen-Bader ! 👋🏻

1. Based on the progressive disclosure nature of being able to set a visibility date, I would say this component would probably not replace the Visible/Hidden setting UI. Here’s a [link](https://shopify.workplace.com/groups/1294395484626415/posts/1321445901921373/?comment_id=1330392557693374&reply_comment_id=1330494334349863) to a WP post with more on that (although I see you already 👍🏻'd it.)
2. As part of this improvement, yes! We also are updating the checkbox documentation to fix current language, like this: “… toggling a setting on or off” — you can find that GH issue [here](https://github.com/Shopify/polaris/pull/8497/commits) where JJ made a commit.
And I agree with your “where it wouldn’t fit” example. In this Pre-chat form setting, making the selection _shouldn’t_ save immediately, and rather we want admin to wait until the merchant has reviewed the messaging (“Opt in to get…”) before making the change/update. As part of the setting toggle behavior, the switch is made upon button click; The merchant does not have to save via the [contextual save bar](https://polaris.shopify.com/components/selection-and-input/contextual-save-bar).

**Comment by Jen-Bader** [comment]
Thanks, @heydouble-a!
1. I saw your WP post after posting here. In Inbox, we don't have a "set visibility date", so was thinking we'd use the "on'off" toggle. 

**Context:** 
Currently, we use "visible" language. 
<img width="271" alt="image" src="https://user-images.githubusercontent.com/55419135/222265660-457fb8c3-8eb2-4980-b735-1751ea993bc6.png">

However, we're now including button options along w/ visibility, since the 2 are related. 
<img width="369" alt="image" src="https://user-images.githubusercontent.com/55419135/222261181-e07b7066-d93b-4ffb-a722-a5f49dce176a.png">
In this scenario, here's how I understand the guideline: If the setting 
 - is dependent on other settings: _I wouldn't call on/off "dependent" on button choice or location, so this doesn't apply_
 - uses progressive disclosure: _I think of progressive disclosure in the way Jakob Nielson defines it, i.e. "a technique that “defers advanced or rarely used features to a secondary screen..." so this doesn't apply_
 - or has options that are not a simple “On”/“Off”: _ I regard this as applying since there are additional options, so would then keep the "visible" language_

Does this maintain the spirit of the guideline's intent? 
If so, the language changes based on what it's grouped with. 
e.g., yesterday we were talking about moving the toggle to the preview that sits at the right of it. If we did this, language would change back to on/off. 
<img width="294" alt="image" src="https://user-images.githubusercontent.com/55419135/222268719-3050eff3-7313-43a9-911c-d095c5ca1b0c.png">


**I'm sharing the above examples to highlight how how terminology might change for similar scenarios, and how this might confuse merchants if we were to make incremental updates (which we're thinking of, given we're now working in shorter cycles and need to limit scope).** 

I do think your proposal brings clarity and am grateful for it. My intent is to share some considerations in the Inbox team's desire to adopt this guideline. cc @jawsmartin 

2. Got it. Thanks for this clarification!

**Comment by giliaangell** [comment]
Hello, a follow up to a nuance around selecting an on/off when there is a call to a database that creates lag time in activation feedback. Our Payments dev Corin McCargue prototyped this for a problem we had when we updated a new email sign up checkbox in a payments card. This feels like the right direction and weight of feedback to be relaying for this action and I'd like to approve it: 

1. 
![image](https://user-images.githubusercontent.com/95946352/222802206-f61ff2b5-d073-47b7-b91a-eccee71d0246.png)

2. 
![image](https://user-images.githubusercontent.com/95946352/222802290-64e24a32-8fc8-48ca-8b28-3243b70bb7f3.png)



3. 
![image](https://user-images.githubusercontent.com/95946352/222802482-c0f8414b-be19-4e03-9af9-9014a03fbdfb.png)


https://videobin.shopify.io/v/qk1gB6

**Comment by jjgali** [comment]
> Hello, a follow up to a nuance around selecting an on/off when there is a call to a database that creates lag time in activation feedback. Our Payments dev Corin McCargue prototyped this for a problem we had when we updated a new email sign up checkbox in a payments card. This feels like the right direction and weight of feedback to be relaying for this action and I'd like to approve it:

Hey @giliaangell! I spoke with a few other designers on Polaris, and we agree that while this solution isn't _incorrect_ usage of a checkbox and toast, there are a few points worth mentioning:
- The biggest issue is that a long delay like this exists for a simple action. I'm sure you've explored ways to improve it, but just want to call out that resolving that delay seems like the best experience for users, if possible.
- Checkboxes are typically used for actions that deliver immediate change. So this might be somewhat jarring for users who are expecting a more immediate response. Testing this experience with users might be a good way to validate the proposed solution.

**Comment by deannatron** [comment]
@jjgali 

>Checkboxes are typically used for actions that deliver immediate change.

Can you clarify what you mean by this piece? I want to make sure you don't mean that checkboxes should invoke instant save instead of contextual save (the save bar at the top of the page)

**Comment by jjgali** [comment]
Hey @deannatron I agree with you on the contextual save bar--poor choice of words, I was more thinking along the lines of an immediate response from that particular element. I wouldn't expecting a checkbox to need to load. Does that make sense?

**Comment by deannatron** [comment]
> Hey @deannatron I agree with you on the contextual save bar--poor choice of words, I was more thinking along the lines of an immediate response from that particular element. I wouldn't expecting a checkbox to need to load. Does that make sense?

Yes makes sense—that's what I thought you meant but wanted to double-check to be certain. Thanks JJ!

**Comment by giliaangell** [comment]
@jjgali Thanks for the comments on the load time on the checkbox. I agree that the varying load times for simple action is not ideal and I've followed up with the devs to learn more about prospects for speeding that up. One of the reasons we landed on the checkbox is that it does double duty as quick edit element and a confirmation of status (receiving email notifications) regardless if the card is open for editing, and isn't a page level save. Indeed it's unexpected for a checkbox to initiate a loader, and while that is (hopefully temporarily) still the case, the latter still clearly communicates that something is happening and the user doesn't need to keep clicking it for it to take. 

We'll keep an eye on it and communicate back here.

**Comment by heydouble-a** [comment]
@ouellettejordan tagging for visibility of the ongoing conversation ^

**Comment by alex-page** [comment]
Hey folks, forwarding my feedback on the PR to here.

Adding opinions like this to our components scales their complexity. If I was going to build a "Setting toggle" today I would not expect to get have a "Setting toggle" as a part of the design system. I would build it with `Card` `button` and typography/layout components. Big red flags for me are adding multiple properties and translations to our components.

I think this would be better suited as a pattern or documentation.

> Hey @oluwatimio. I am not aligned with these changes. You are adding a lot of props and translations to the component. Polaris should focus on creating small composable patterns. Additional props is more maintenance, design opinions, documentation and examples.
> 
> I think this component would be better outside of Polaris. We have recently been figuring out patterns and I think this is a good example of what a pattern could be cc @sophschneider and @johanstromqvist.

https://github.com/Shopify/polaris/pull/8547#pullrequestreview-1336090611

**Comment by jawsmartin** [comment]
Hello!! As Jennifer mentioned above, the Messaging team is looking to adopt this Settings toggle in our [Admin UX Improvements vault project](https://vault.shopify.io/gsd/projects/32012). 

Is it a rule that the status indicator badge must be `neutral` when `On`? At a glance, it's hard to parse what is `On` and `Off` in our channel when both states are `neutral`. The `success` green for `On` badge feels quite powerful for parsing a page. It seems like it instills confidence items are going as expected.

In our sales channel, we will be using this settings toggle in 4 places:

• Indicating when Chat is` On/Off`
• Indicating when a specific messaging channel is `On/Off`
• Indicating when an automatic greeting is `On/Off`
• Indicating when an Instant answer is `On/Off`

<img width="1645" alt="Screenshot 2023-03-14 at 12 59 02 PM" src="https://user-images.githubusercontent.com/3891708/225095956-e27ecb3c-f669-4af3-a131-8ed37ddc6617.png">

<img width="1180" alt="Screenshot 2023-03-14 at 12 51 20 PM" src="https://user-images.githubusercontent.com/3891708/225095975-e496b2bd-0e76-4d87-bd00-d62bac30bd02.png">

<img width="1186" alt="Screenshot 2023-03-14 at 12 51 23 PM" src="https://user-images.githubusercontent.com/3891708/225096005-71d0cf03-5d8e-40e0-99b4-fe6fd1b12f73.png">

<img width="1916" alt="Screenshot 2023-03-14 at 12 58 39 PM" src="https://user-images.githubusercontent.com/3891708/225096027-7f3338dc-863e-4087-aa00-22088d2a526a.png">

**Comment by heydouble-a** [comment]
@jawsmartin 
The `On` badge is not required to be gray, no. I agree—green makes the status much clearer and scannable.
Excited that your team will be utilizing this component!

**Comment by jawsmartin** [comment]
When will the Settings toggle component be updated in Polaris and available to use? 

Right now, we are building the component from small Polaris components to look like this new settings toggle.

My guess is that this is OK, and we will ultimately replace these when the new settings toggle is introduced.

cc @raderj89 @chukitow
